src/ImGui.NET.SampleProgram/MemoryEditor.cs

```diff
@@ -0,0 +1,271 @@
+                {
+                    ImGui.SameLine(line_start_x + cell_width * n);
+
+                    if (DataEditingAddr == addr)
+                    {
+                        // Display text input on current byte
+                        ImGui.PushID(addr);
+
+                        // FIXME: We should have a way to retrieve the text edit cursor position more easily in the API, this is rather tedious.
+                        TextEditCallback callback = (data) =>
+                        {
+                            int* p_cursor_pos = (int*)data->UserData;
+                            if (!data->HasSelection())
+                                *p_cursor_pos = data->CursorPos;
+                            return 0;
+                        };
+                        int cursor_pos = -1;
+                        bool data_write = false;
+                        if (DataEditingTakeFocus)
+                        {
+                            ImGui.SetKeyboardFocusHere();
+                            ReplaceChars(DataInput, FixedHex(mem_data[addr], 2));
+                            ReplaceChars(AddrInput, FixedHex(base_display_addr + addr, addr_digits_count));
+                        }
+                        ImGui.PushItemWidth(ImGui.GetTextSize("FF").X);
+
+                        var flags = InputTextFlags.CharsHexadecimal | InputTextFlags.EnterReturnsTrue | InputTextFlags.AutoSelectAll | InputTextFlags.NoHorizontalScroll | InputTextFlags.AlwaysInsertMode | InputTextFlags.CallbackAlways;
+                        if (ImGui.InputText("##data", DataInput, 32, flags, callback, new IntPtr(&cursor_pos)))
+                            data_write = data_next = true;
+                        else if (!DataEditingTakeFocus && !ImGui.IsLastItemActive())
+                            DataEditingAddr = -1;
+
+                        DataEditingTakeFocus = false;
+                        ImGui.PopItemWidth();
+                        if (cursor_pos >= 2)
+                            data_write = data_next = true;
+                        if (data_write)
+                        {
+                            int data;
+                            if (TryHexParse(DataInput, out data))
+                                 mem_data[addr] = (byte)data;
+                        }
+                        ImGui.PopID();
+                    }
+                    else
+                    {
+                        ImGui.Text(FixedHex(mem_data[addr], 2));
+                        if (AllowEdits && ImGui.IsLastItemHovered() && ImGui.IsMouseClicked(0))
+                        {
+                            DataEditingTakeFocus = true;
+                            DataEditingAddr = addr;
+                        }
+                    }
+                }
+
+                ImGui.SameLine(line_start_x + cell_width * Rows + glyph_width * 2);
+                //separator line drawing replaced by printing a pipe char
```

Comment:
I didn't want to get too low-level right from the start :D As a bonus, it now looks more like TotalCmd's hex viewer.